### ISSUE TITLE
Labels limit & Jumping Lanes

### ISSUE THREAD

**Issue by headwinds** [lflores/gravity-bubbles#1]
I've been experimenting with this codebase and I'd like to contribute to this project by possibly adding a few things to how the chart is configured.

For instance, I would like to see labels within the bubbles up until a certain radius limit. Perhaps, you already have this?! Thus I could work on documentation on the wiki and simply start asking questions. 

I started this [codepen](http://codepen.io/headwinds/pen/xVYbZV). I want to show data separated into channels. I'd also like show how data could move from one to channel to another over time through animation. 

So... 
1. adding label limits (should be easy eh?)
2. data swimming upstream ( probably will take a lot of experimenting) 

I'm happy to help out and I really like what you've started here.

**Comment by lflores** [comment]
Thank you for your feedback, and your codepen looks amazing, it's best example of the use of gravity bubbles.
About your doubts.
1. Labels as you wish, are running on last version of project and you can see on this [fiddle](https://jsfiddle.net/leoflores/6cLpuL7j/) please let me know if you can see (i'm fiddle rookie).Basically the label has same syntax of tooltip, if you use "{property name}" string as template, the component will try to extract value of name property of item data. The labels are auto show depending on width of text and diameter of bubble.Also label can do an split if you use a new line "{name}\n{description}" to take advantage of available space. I hope to do same with spaces, dash, underscore, to apply same logic to more complex texts. Let me know your thoughts.
2. I do similar thing with two approaches, doing an asynchronous update of data. And for the other hand, I have experimented with set of data on code, separated by months and years. Each time that selector changed I changed set of data collection. Also I updated by timeout the data set, to animate evolution of data. If I find the example I will share it.
The documentation is my great debt, please tell me if something is unclear
Thank you for your offer, we'll keep in contact

**Comment by lflores** [comment]
Viewing your codepen, I have detected an error with group name length, when you do a grouping by channel. "Autonomous" is too large to see in group header. For this I added a new Issue to handle it #2
Thanks

**Comment by headwinds** [comment]
Yep, I can see your fiddle... I'll take a look... what I'd like to see if labels like my other [codepen](http://codepen.io/headwinds/pen/lHaqI) - I needed to fit long strings which could be 30 or 40 characters.

I may need a take word like "autonomous" and break it up into "Autono-" + "mous" but ideally I'd like it to fit.

**Comment by headwinds** [comment]
Can you create a sample of your progress?

I'll also try to cook up a prototype on codepen. I want to show a bubble that a channel property like "Web, Chat, Autonomous". Then it the bubble would animate through the channels so that the viewer could see it started in Web and ended up in Autonomous. I wouldn't want them all to animate at once but if they animated one at a time if might be effective and easy to follow, and the user could select a new bubble to follow or stop the presentation at any time.  

Let's start the documentation on the wiki...

**Comment by lflores** [comment]
Well, nowadays is not supported. The code today just try to fit in enabled space, if not hide the label. It's a good enhancement. I added #3 issue.

**Comment by lflores** [comment]
Hi @headwinds, I tried to do an example to explain transitions in better manner. I do an new example called evolution, and you can test it in [test site](http://lflores.github.io/evolution/).
Testing the example, I discovered that transitions don't look good when data changes, and I'm working on this, but I want to share the example. Please tell me if it helps you.
Regards.

**Comment by headwinds** [comment]
I like where you are going with this experiment.

As you switch months, instead of a redrawing the bubbles each time, can you animate the change so that we see some bubbles grow while other shrink? We want to see the evolution and measure the change.

For instance, Vis moves from 45% (January) to 49% (February) to 45% (March). We should see that bubble's radius animate. 

Also, instead of buttons, how about using a range slider for the months? Scrubbing the timeline would be cool.

**Comment by lflores** [comment]
Hi @headwinds, I have updated the component to version 1.0.1, with new features.
- I have added a new property to auto fit the labels, as you wish. You can see the [fiddle](https://jsfiddle.net/leoflores/6cLpuL7j/) or [demo](http://lflores.github.io/evolution/). Basically it's a new configuration property `{data:{label:{autofit:true}}}`
- Also I have worked to keep data and bubbles positions between asynchronous calls (It was a hard work ;), scanning documentation and debugging). Now the transitions between request are continuous, and it was the point that I told you "don't look good", but now I thinks that is solved. 
- I have modified you own "martian" example and I added a switch to common label or auto fit use. 
- Now you have a new project to host examples called [lflores.github.com](https://github.com/lflores/lflores.github.com) where you can add your own folder (martian) that will be hosted under lflores.github.com/martian to browse example.
- And I do little changes with styles, such as un selectable texts, I solved a conflict with label class between bubbles and bootstrap, and some little naming changes, to avoid posibles conflicts 
- I leave the slider requested for the next release.

Please give me your feedback and thoughts. See you

**Comment by headwinds** [comment]
awesome! this is great work and really captures that cool factor from the Obama example. I'll grab the latest and check out what you've done in detail this weekend.

**Comment by lflores** [comment]
Hi @headwinds, I have worked in the "evolution" example and [here](http://lflores.github.io/evolution/) you have a new version with slider as you wishes.
I have used [this](https://github.com/seiyria/bootstrap-slider) component that is a bootstrap slider.
I'm lost in all requirements,but I think that the only thing that we need is the best group label rules.
That requirement is in another issue (#2), then I think that if you agree, we can close this.
